peers_base.js

```diff
@@ -68,15 +68,46 @@ TChannelPeersBase.prototype.close = function close(peers, callback) {
     }
 };
 
-TChannelPeersBase.prototype.sanitySweep = function sanitySweep() {
+TChannelPeersBase.prototype.sanitySweep =
+function sanitySweep(callback) {
     var self = this;
 
-    var peers = self.values();
-    for (var i = 0; i < peers.length; i++) {
+    nextPeer(self.values(), 0, callback);
```

Comment:
Dear reviewers: please fact check my on iterating a list that may mutate while I iterate it.